onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -155,19 +198,32 @@ def change_password(self, request, *args, **kwargs):  # noqa
         user_profile = self.get_object()
         current_password = request.data.get('current_password', None)
         new_password = request.data.get('new_password', None)
+        lock_out = check_user_lockout(request)
 
         if new_password:
-            if user_profile.user.check_password(current_password):
-                metadata = user_profile.metadata or {}
-                metadata['last_password_edit'] = timezone.now().isoformat()
-                user_profile.user.set_password(new_password)
-                user_profile.metadata = metadata
-                user_profile.user.save()
-                user_profile.save()
-
-                return Response(status=status.HTTP_204_NO_CONTENT)
-
-        return Response(status=status.HTTP_400_BAD_REQUEST)
+            if not lock_out:
+                if user_profile.user.check_password(current_password):
+                    metadata = user_profile.metadata or {}
+                    metadata['last_password_edit'] = timezone.now().isoformat()
+                    user_profile.user.set_password(new_password)
+                    user_profile.metadata = metadata
+                    user_profile.user.save()
+                    user_profile.save()
+
+                    return Response(status=status.HTTP_204_NO_CONTENT)
+                else:
```

Comment:
We could also do without the else section since we have a return in the above if block.